internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,398 @@
+				mu sync.Mutex
+			)
+
+			log.Trace("starting job stream for jobs", "total_jobs", len(triggerJobs))
+			wg.Add(len(triggerJobs))
+
+			// NOTE(briancain): This loop starts N goroutines concurrently for
+			// each trigger job to stream back to the requester.
+			// TODO(briancain): turn the http errors into Messages to send back
+			for _, jId := range triggerJobs {
+				go func(jId string) {
+					defer wg.Done()
+
+					stream, err := client.GetJobStream(ctx, &pb.GetJobStreamRequest{
+						JobId: jId,
+					})
+					if err != nil {
+						log.Error("server failed to get job stream output for trigger", "job_id", jId, "err", err)
+						http.Error(w,
+							fmt.Sprintf("server failed to obtain job stream output: %s", err),
+							http.StatusInternalServerError)
+						return
+					}
+
+					// Wait for open confirmation
+					resp, err := stream.Recv()
+					if err != nil {
+						log.Error("server failed to stream job output", "err", err)
+						http.Error(w,
+							fmt.Sprintf("server failed to receive job stream output: %s", err),
+							http.StatusInternalServerError)
+						return
+					}
+					if _, ok := resp.Event.(*pb.GetJobStreamResponse_Open_); !ok {
+						log.Error("server failed to open job stream output, got unexpected message", "event", resp.Event)
+						http.Error(w,
+							fmt.Sprintf("job stream failed to open, got unexpected message: %T", resp.Event),
+							http.StatusInternalServerError)
+						return
+					}
+
+					var (
+						jobComplete bool
+						exitCode    string
+					)
+
+					log.Trace("reading job stream for job", "job_id", jId)
+
+					// read and send the stream
+					for {
+						resp, err := stream.Recv()
+						if err != nil {
+							http.Error(w,
+								fmt.Sprintf("server failed to receive job stream output: %s", err),
+								http.StatusInternalServerError)
+							return
+						}
+						if resp == nil {
+							// This shouldn't happen, but if it does, just ignore it.
+							log.Warn("nil response received, ignoring")
+							continue
+						}
+
+						select {
+						case <-cn.CloseNotify():
+							log.Trace("client closed connection to stream")
+							return
+						default:
+							// Get jobstream output and return string chunk message back
+							time.Sleep(time.Second)
+
+							// the message to return
+							var value interface{}
+							var valueType string
+
+							switch event := resp.Event.(type) {
+							case *pb.GetJobStreamResponse_Complete_:
+								jobComplete = true
+								valueType = "Complete"
+
+								if event.Complete.Error == nil {
+									log.Info("job completed successfully")
+									exitCode = "0"
+								} else {
+									exitCode = "1"
+									st := status.FromProto(event.Complete.Error)
+									log.Warn("job failed", "code", st.Code(), "message", st.Message())
+									http.Error(w,
+										fmt.Sprintf("job failed to complete: job code %s: %s", st.Code(), st.Message()),
+										http.StatusInternalServerError)
+								}
```

Comment:
Question, do we need to return here? Either with completed or the error block?

Reply:
One sec, I have a commit that will fix this up 😄

Reply:
I've updated this func locally to instead return the error directly in Message rather than an http error.